Mathlib/Topology/SubsetProperties.lean

```diff
@@ -1307,6 +1309,119 @@ theorem IsClosed.exists_minimal_nonempty_closed_subset [CompactSpace α] {S : Se
+      -- ... but f i is a set in S, so x ∈ ⋃ S as well.
+      exact ⟨f i, Subtype.mem (f i), hi⟩
+    · -- Suppose x ∈ ⋃ s, then x ∈ s for some s ∈ S.
+      intro x hx
+      rw [mem_sUnion] at hx
+      rcases hx with ⟨s, h, hxs⟩
+      -- Choose n with f n = s (using surjectivity of f).
+      have : ∃ n, f n = s := by
+        obtain ⟨y, hy⟩ := hf ⟨s, h⟩
+        use y
+        simp_all only [Subtype.forall]
+      rcases this with ⟨n, hn⟩
+      exact ⟨f n, mem_range_self n, (by rw [hn]; exact hxs)⟩
+
+/-- Countable unions of σ-compact sets are σ-compact. -/
+lemma isSigmaCompact_of_countable_sigma_compact (S : Set (Set α)) (hc : Set.Countable S)
+    (hcomp : ∀ s : S, IsSigmaCompact s (α := α)) : IsSigmaCompact (⋃₀ S) := by
+  -- Choose a decomposition s = ⋃ s_i for each s ∈ S.
+  choose K hcomp hcov using fun s ↦ hcomp s
+  -- Then, we have a countable union of countable unions of compact sets, i.e. countably many.
+  have : ⋃₀ S = ⋃₀ range (K.uncurry) := by
+    calc ⋃₀ S
+      _ = ⋃ s : S, s := by rw [← sUnion_eq_iUnion]
+      _ = ⋃ s : S, ⋃ n, (K s n) := by simp_rw [hcov]
+      _ = ⋃ s : S, ⋃ n, (K.uncurry ⟨s, n⟩) := by rw [Function.uncurry_def]
+      _ = ⋃ (s : S) (n : ℕ), (K.uncurry ⟨s, n⟩) := by rw [iUnion_coe_set]
+      _ = ⋃₀ range (K.uncurry) := by rw [← iUnion_prod', sUnion_range]
+  rw [this]
+  rw [← countable_coe_iff] at hc
+  refine isSigmaCompact_of_countable_compact (countable_range (K.uncurry)) fun s hs ↦ ?_
+  obtain ⟨⟨ys, yn⟩, hy⟩ := mem_range.mp hs
+  rw [← hy]
+  exact hcomp ys yn
+
+/-- A closed subset of a σ-compact set is σ-compact. -/
+lemma IsSigmaCompact.of_isClosed_subset {s t : Set α} (ht : IsSigmaCompact t)
+    (hs : IsClosed s) (h : s ⊆ t) : IsSigmaCompact s := by
+  rcases ht with ⟨K, hcompact, hcov⟩
+  refine ⟨(fun n ↦ s ∩ (K n)), fun n ↦ (hcompact n).inter_left hs, ?_⟩
+  rw [← inter_iUnion, hcov]
+  exact inter_eq_left.mpr h
+
+/-- If `s` is σ-compact and `f` continuous on a set `w` containing `s`, `f(s)` is σ-compact.-/
+lemma IsSigmaCompact.image_of_continuousOn {f : α → β} {s : Set α} (hs : IsSigmaCompact s)
+    (hf : ContinuousOn f s) : IsSigmaCompact (f '' s) := by
+  rcases hs with ⟨K, hcompact, hcov⟩
+  refine ⟨fun n ↦ f '' K n, ?_, (by rw [← hcov]; exact image_iUnion.symm)⟩
+  refine fun n ↦ (hcompact n).image_of_continuousOn (hf.mono ?_)
+  rw [← hcov]
+  exact subset_iUnion K n
+
+/-- If `s` is σ-compact and `f` continuous, `f(s)` is σ-compact. -/
+lemma IsSigmaCompact.image {f : α → β} (hf : Continuous f) {s : Set α} (hs : IsSigmaCompact s) :
+    IsSigmaCompact (f '' s) := hs.image_of_continuousOn hf.continuousOn
+
+/-- If `f : X → Y` is an `Embedding`, the image `f '' s` of a set `s` is σ-compact
+if and only `s`` is σ-compact.
+This does not hold for merely inducing maps; direction `←` requires injectivity. -/
+lemma Embedding.isSigmaCompact_iff_isSigmaCompact_image {f : α → β} {s : Set α}
+    (hf : Embedding f) : IsSigmaCompact s ↔ IsSigmaCompact (f '' s) := by
```

Comment:
This looks better in reverse to me:
```suggestion
lemma Embedding.isSigmaCompact_image_iff {f : α → β} {s : Set α}
    (hf : Embedding f) : IsSigmaCompact (f '' s) ↔ IsSigmaCompact s := by
```
(obviously you'll have to tweak the proof too)

Reply:
I chose the name to mirror `Embedding.isCompact_iff_isCompact_image`. Should I change that one as well?

Reply:
That's a good reason for the current name; certainly I agree they should match.

Let's see if @urkud or @sgouezel agree with renaming (and flipping) both.

Reply:
These names come from https://github.com/leanprover-community/mathlib/pull/367, so are very old and possibly before conventions were well established!

Reply:
This should just be called `Embedding.is[Sigma]Compact_iff`, by analogy with [Inducing.isCompact_iff](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Topology/SubsetProperties.html#Inducing.isCompact_iff). The rationale is the following: most of the embeddings we consider go from a space we want to understand to a space we understand well, so this should just be thought of as a criterion for compacity on the domain, which is why we don't emphasize the image part.

Reply:
Makes sense to me, renamed. (Indeed, counting all current occurrences in mathlib, I see one where we rewrite "forward" and five where we rewrite backwards.)